Packet provider crash

### Provider Version
1.2

### Terraform Version
0.11.2

### Affected Resource(s)
Please list the resources as a list, for example:
- packet_project
- packet_ssh_key

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
I'm following along from https://www.terraform.io/docs/providers/packet/index.html
```
$ cat main.tf 
variable "packet_api_key" {}
variable "packet_project_name" {}

terraform {
  required_version = ">= 0.11.2"
}

provider "packet" {
  auth_token = "${var.packet_api_key}"
}

module "packet_server" {
  source       = "./modules/packet.net/"
  project_name = "${var.packet_project_name}"
}

output "project_name" {
  value = "${var.packet_project_name}"
}

output "project_id" {
  value = "${module.packet_server.project_id}"
}
```

```
$ cat modules/packet.net/packet.net.tf 
variable "project_name" {}

resource "packet_project" "phil" {
  name = "${var.project_name}"
  payment_method = "test"          # Only required for a non-default payment method
  #payment_method = "Phil Was Here"          # Only required for a non-default payment method
}

resource "packet_ssh_key" "key_phil" {
  name       = "pgporada"
  public_key = "${file("${path.module}/../../ssh_keys/packet_pgporada.pub")}"
}

output "project_id" {
  value = "${packet_project.phil.id}"
}

output "project_payment_method" {
  value = "${packet_project.phil.payment_method}"
}
```

### Panic Output
https://gist.github.com/pgporada/c888812d378f8b45553cada9660954aa for payment method of "Phil Was Here"
https://gist.github.com/pgporada/2038ed44fcd62b77a72924a7cbcebbbe for payment method of "test"

I am unsure what the payment method should be if it's not the default. Additionally I get an error, but no crash, if I try to output the payment method when using the default.
```
$ terraform apply -var-file environments/test/test.tfvars
packet_ssh_key.key_phil: Refreshing state... (ID: efc554a2-1b1a-4248-abb6-0c1f7af334f5)

Error: Error running plan: 1 error(s) occurred:

* module.packet_server.output.project_payment_method: Resource 'packet_project.phil' does not have attribute 'payment_method' for variable 'packet_project.phil.payment_method'
```

### Expected Behavior
Not to see a crash

### Actual Behavior
I saw a crash because I entered the payment method as "Phil Was Here". Using "test" will also result in a crash.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
```
$ terraform plan -var-file environments/test/test.tfvars 
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

packet_ssh_key.key_phil: Refreshing state... (ID: efc554a2-1b1a-4248-abb6-0c1f7af334f5)
packet_project.phil: Refreshing state... (ID: 44c6ee41-daaf-486a-b47c-3e9df14c6bf4)

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.packet_server.packet_project.phil
      payment_method: "" => "Phil Was Here"


Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
```
```
$ terraform apply -var-file environments/test/test.tfvars
packet_ssh_key.key_phil: Refreshing state... (ID: efc554a2-1b1a-4248-abb6-0c1f7af334f5)
packet_project.phil: Refreshing state... (ID: 44c6ee41-daaf-486a-b47c-3e9df14c6bf4)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.packet_server.packet_project.phil
      payment_method: "" => "Phil Was Here"


Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.packet_server.packet_project.phil: Modifying... (ID: 44c6ee41-daaf-486a-b47c-3e9df14c6bf4)
  payment_method: "" => "Phil Was Here"

Error: Error applying plan:

1 error(s) occurred:

* module.packet_server.packet_project.phil: 1 error(s) occurred:

* packet_project.phil: unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


panic: interface conversion: interface {} is nil, not string
2018-01-25T15:16:26.783-0500 [DEBUG] plugin.terraform-provider-packet_v1.2.0_x4: 
2018-01-25T15:16:26.783-0500 [DEBUG] plugin.terraform-provider-packet_v1.2.0_x4: goroutine 103 [running]:
2018-01-25T15:16:26.783-0500 [DEBUG] plugin.terraform-provider-packet_v1.2.0_x4: github.com/terraform-providers/terraform-provider-packet/packet.resourcePacketProjectUpdate(0xc42017aaf0, 0xb857a0, 0xc4203cb540, 0x24, 0x1165480)
2018-01-25T15:16:26.783-0500 [DEBUG] plugin.terraform-provider-packet_v1.2.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-packet/packet/resource_packet_project.go:86 +0x37d
2018-01-25T15:16:26.783-0500 [DEBUG] plugin.terraform-
```

--- a/packet/resource_packet_project.go
+++ b/packet/resource_packet_project.go
@@ -1,9 +1,16 @@
 package packet
 
 import (
+	"path"
+	"regexp"
+	"strings"
+
 	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
 	"github.com/packethost/packngo"
 )
+
+var uuidRE = regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
 
 func resourcePacketProject() *schema.Resource {
 	return &schema.Resource{
@@ -26,6 +33,15 @@ func resourcePacketProject() *schema.Resource {
 			"updated": &schema.Schema{
 				Type:     schema.TypeString,
 				Computed: true,
+			},
+			"payment_method_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return strings.ToLower(strings.Trim(old, `"`)) == strings.ToLower(strings.Trim(new, `"`))
+				},
+				ValidateFunc: validation.StringMatch(uuidRE, "must be a valid UUID"),
 			},
 		},
 	}
@@ -51,7 +67,7 @@ func resourcePacketProjectCreate(d *schema.ResourceData, meta interface{}) error
 func resourcePacketProjectRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*packngo.Client)
 
-	key, _, err := client.Projects.Get(d.Id())
+	proj, _, err := client.Projects.Get(d.Id())
 	if err != nil {
 		err = friendlyError(err)
 
@@ -65,22 +81,24 @@ func resourcePacketProjectRead(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	d.Set("id", key.ID)
-	d.Set("name", key.Name)
-	d.Set("created", key.Created)
-	d.Set("updated", key.Updated)
+	d.Set("id", proj.ID)
+	d.Set("payment_method_id", path.Base(proj.PaymentMethod.URL))
+	d.Set("name", proj.Name)
+	d.Set("created", proj.Created)
+	d.Set("updated", proj.Updated)
 
 	return nil
 }
 
 func resourcePacketProjectUpdate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*packngo.Client)
-
-	updateRequest := &packngo.ProjectUpdateRequest{
-		ID:   d.Get("id").(string),
-		Name: d.Get("name").(string),
+	updateRequest := &packngo.ProjectUpdateRequest{ID: d.Id()}
+	if d.HasChange("name") {
+		updateRequest.Name = d.Get("name").(string)
 	}
-
+	if d.HasChange("payment_method_id") {
+		updateRequest.PaymentMethodID = d.Get("payment_method_id").(string)
+	}
 	_, _, err := client.Projects.Update(updateRequest)
 	if err != nil {
 		return friendlyError(err)

--- a/packet/resource_packet_project_test.go
+++ b/packet/resource_packet_project_test.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/hashicorp/terraform/helper/acctest"
 	"github.com/hashicorp/terraform/helper/resource"
 	"github.com/hashicorp/terraform/terraform"
 	"github.com/packethost/packngo"
@@ -11,6 +12,7 @@ import (
 
 func TestAccPacketProject_Basic(t *testing.T) {
 	var project packngo.Project
+	rInt := acctest.RandInt()
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
@@ -18,18 +20,45 @@ func TestAccPacketProject_Basic(t *testing.T) {
 		CheckDestroy: testAccCheckPacketProjectDestroy,
 		Steps: []resource.TestStep{
 			resource.TestStep{
-				Config: testAccCheckPacketProjectConfig_basic,
+				Config: testAccCheckPacketProjectConfig_basic(rInt),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckPacketProjectExists("packet_project.foobar", &project),
-					testAccCheckPacketProjectAttributes(&project),
 					resource.TestCheckResourceAttr(
-						"packet_project.foobar", "name", "foobar"),
+						"packet_project.foobar", "name", fmt.Sprintf("foobar-%d", rInt)),
 				),
 			},
 		},
 	})
 }
 
+func TestAccPacketProject_Update(t *testing.T) {
+	var project packngo.Project
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckPacketProjectDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckPacketProjectConfig_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPacketProjectExists("packet_project.foobar", &project),
+					resource.TestCheckResourceAttr(
+						"packet_project.foobar", "name", fmt.Sprintf("foobar-%d", rInt)),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCheckPacketProjectConfig_basic(rInt + 1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPacketProjectExists("packet_project.foobar", &project),
+					resource.TestCheckResourceAttr(
+						"packet_project.foobar", "name", fmt.Sprintf("foobar-%d", rInt+1)),
+				),
+			},
+		},
+	})
+}
 func testAccCheckPacketProjectDestroy(s *terraform.State) error {
 	client := testAccProvider.Meta().(*packngo.Client)
 
@@ -43,15 +72,6 @@ func testAccCheckPacketProjectDestroy(s *terraform.State) error {
 	}
 
 	return nil
-}
-
-func testAccCheckPacketProjectAttributes(project *packngo.Project) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		if project.Name != "foobar" {
-			return fmt.Errorf("Bad name: %s", project.Name)
-		}
-		return nil
-	}
 }
 
 func testAccCheckPacketProjectExists(n string, project *packngo.Project) resource.TestCheckFunc {
@@ -80,7 +100,10 @@ func testAccCheckPacketProjectExists(n string, project *packngo.Project) resourc
 	}
 }
 
-var testAccCheckPacketProjectConfig_basic = fmt.Sprintf(`
+func testAccCheckPacketProjectConfig_basic(r int) string {
+
+	return fmt.Sprintf(`
 resource "packet_project" "foobar" {
-    name = "foobar"
-}`)
+    name = "foobar-%d"
+}`, r)
+}

--- a/website/docs/r/project.html.markdown
+++ b/website/docs/r/project.html.markdown
@@ -25,11 +25,13 @@ resource "packet_project" "tf_project_1" {
 The following arguments are supported:
 
 * `name` - (Required) The name of the Project on Packet.net
+* `payment_method_id` - The UUID of payment method for this project. If you keep it empty, Packet API will pick your default Payment Method.
 
 ## Attributes Reference
 
 The following attributes are exported:
 
 * `id` - The unique ID of the project
+* `payment_method_id` - The UUID of payment method for this project.
 * `created` - The timestamp for when the Project was created
 * `updated` - The timestamp for the last time the Project was updated

